### PR TITLE
Adds the M46 'Bolo' Tactical Dagger

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Adds the M46 'Bolo' Tactical Dagger. The M46 is a medium-sized middle-ground between a bayonet and a machete, dealing intermediate damage with a faster attack speed than either. Intended for marines who cannot fit a machete with their kit but still want a resin-clearing weapon. Can be found in Req or in equipment vendors.

The Bolo can be attached to USCM weapons, but its large size and weight noticeably worsen handling and stability.

Reworks how Melee on resin structures is handled, see comment in the defines file, necessary to make sure the Bolo wasn't overpowered or underpowered. Of note, machetes now break resin doors in two hits, thick doors on three, and weak membranes in one hit.

Adds a variable for resin melee multiplier. Reduced sticky/fast resin and spikes health to 25 and gave them no melee multiplier.

image

Bolo on the left (ours) bayonet on the right

# Explain why it's good for the game

1. It adds a new option for marines to pick to customize their loadout, without being too ephemeral.
2. It's reasonably realistic - militaries do use bolo knives for clearing vegetation.
3. It's cool and doesn't negatively impact anyone's experience.
4. It allows those who want to get something like a machete but don't feel like sacrificing the 3 most important storage slots in a marine's loadout.

The resin damage rework was unfortunately a necessity as otherwise it would have been nigh impossible to design the Bolo in a way that wouldn't have been OP or UP.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
Carlarc, XenocidalEnragement
imageadd: Adds the M46 'Bolo' Tactical Dagger. The M46 is a medium-sized middle-ground between a bayonet and a machete, dealing intermediate damage with a faster attack speed than either. Intended for marines who cannot fit a machete with their kit but still want a resin-clearing weapon. Can be found in Req or in equipment vendors.

add: The Bolo can be attached to USCM weapons, but its large size and weight noticeably worsen handling and stability.

balance: Reworks how Melee on resin structures is handled, see comment in the defines file, necessary to make sure the Bolo wasn't overpowered or underpowered. Of note, machetes now break resin doors in two hits, thick doors on three, and weak membranes in one hit.

code: Adds a variable for resin melee multiplier. Reduced sticky/fast resin and spikes health to 25 and gave them no melee multiplier.

/:cl:

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -360,7 +360,7 @@
 /obj/item/clothing/shoes/attackby(obj/item/I, mob/living/M)
 	if(items_allowed && items_allowed.len)
 		for (var/i in items_allowed)
-			if(istype(I, i))
+			if(istype(I, i) && I.w_class <= maximum_size)
```
[the remark this answers is not in the document]
Comment (carlarctg) [reply code/modules/clothing/clothing.dm:363]: wtf it copmpiled

**Review by morrowwolf — CHANGES_REQUESTED** [review]
I don't think this is necessary for standard USCM usage.

Maybe refit the sprite into a marsoc or other usage?

File: code/modules/cm_aliens/XenoStructures.dm [hunk 2]
```diff
@@ -19,6 +19,7 @@
 	anchored = TRUE
 	health = 200
 	unacidable = TRUE
+	var/melee_multiplier = RESIN_MELEE_DAMAGE_MULTIPLIER
```
Comment (Zonespace27) [inline code/modules/cm_aliens/XenoStructures.dm:22]: code doc

File: code/modules/projectiles/gun_attachables.dm [hunk 3]
```diff
@@ -321,6 +321,41 @@ Defined in conflicts.dm of the #defines folder.
...
+	hitsound = 'sound/weapons/slice.ogg'
+	attack_speed = 7
+	attach_icon = "bolo_knife_a"
+	pixel_shift_x = 14
+	pixel_shift_y = 18
+	hud_offset_mod = -4
+	pry_delay = 1 SECONDS
+
+/obj/item/attachable/bayonet/bolo_knife/New()
+	..()
```
Comment (Zonespace27) [inline code/modules/projectiles/gun_attachables.dm:343]:
```suggestion
	. = ..()
```

File: code/modules/projectiles/gun_attachables.dm [hunk 4]
```diff
@@ -321,6 +321,41 @@ Defined in conflicts.dm of the #defines folder.
 	..()
 	accuracy_unwielded_mod = -HIT_ACCURACY_MULT_TIER_1
 
+/obj/item/attachable/bayonet/bolo_knife
+	name = "\improper M46 'Bolo' Tactical Dagger"
+	desc = "A rarely issued, medium-sized tactical dagger that can be seen as the middle-ground between a bayonet and a machete. Unlike the latter, it can be attached to guns for a stronger prying and stabbing tip, at the cost of handling and stability. Its frame allows for quick stabbing motions, faster than either of its counterparts."
+	desc_lore = "This bolo knife is not often seen on the field, and is mostly issued to troops in operations which would benefit from a machete to clear vegetation but due to their specialization cannot make the space for the bulky piece of equipment. It features a ripping edge on its backside that's useful for chopping up firewood or aircraft fuselage, and a neat compass on its pommel that seems to be broken, unfortunately."
```
Comment (Zonespace27) [inline code/modules/projectiles/gun_attachables.dm:327]: is "bolo" a proper noun?

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Review by nauticall — CHANGES_REQUESTED** [review]
imho the bolo knife looks WAY too similar to the preexisting bayonet that im not really comfortable with it as-is. try changing the color palette of it to be different (more blue, perhaps?) and also change the shape so it doesn't just look like an oversized knife. I get that you (or whoever sprited it) wanted to emulate a sort of "curve" but it doesn't look like that went far enough for it to be noticeable.

consider something like curving the _edge_ more, like this bolo:

image

**Pull request closed without merging**